<?xml version="1.0" ?>
<robot name="dual_arms_with_environment">

    <end_effector name="irb120_eef" parent_link="tool0" group="irb120_arm" parent_group="irb120_arm"/>
    <end_effector name="ar4_eef" parent_link="ar4_ee_link" group="ar_manipulator" parent_group="ar_manipulator"/>

    <group name="irb120_arm"><joint name="joint_1"/><joint name="joint_2"/><joint name="joint_3"/><joint name="joint_4"/><joint name="joint_5"/><joint name="joint_6"/></group>
    <group name="ar_manipulator"><joint name="ar4_joint_1"/><joint name="ar4_joint_2"/><joint name="ar4_joint_3"/><joint name="ar4_joint_4"/><joint name="ar4_joint_5"/><joint name="ar4_joint_6"/></group>
    <group name="ar_gripper"><joint name="ar4_gripper_jaw1_joint"/></group>
    <group name="irb120_gripper"><joint name="gripper_ABB_Gripper_Finger_1_Joint"/></group>
    <group name="dual_arms"><group name="ar_manipulator"/><group name="irb120_arm"/></group>

    <group_state group="irb120_arm" name="home"><joint name="joint_1" value="0"/><joint name="joint_2" value="0"/><joint name="joint_3" value="0"/><joint name="joint_4" value="0"/><joint name="joint_5" value="1.5708"/><joint name="joint_6" value="0"/></group_state>
    <group_state group="ar_manipulator" name="home"><joint name="ar4_joint_1" value="0"/><joint name="ar4_joint_2" value="0"/><joint name="ar4_joint_3" value="0"/><joint name="ar4_joint_4" value="0"/><joint name="ar4_joint_5" value="0"/><joint name="ar4_joint_6" value="0"/></group_state>
    
    <disable_collisions link1="base_link" link2="link_1" reason="Adjacent" />
    <disable_collisions link1="link_1" link2="link_2" reason="Adjacent" />
    <disable_collisions link1="link_2" link2="link_3" reason="Adjacent" />
    <disable_collisions link1="link_3" link2="link_4" reason="Adjacent" />
    <disable_collisions link1="link_4" link2="link_5" reason="Adjacent" />
    <disable_collisions link1="link_5" link2="link_6" reason="Adjacent" />
    <disable_collisions link1="ar4_base_link" link2="ar4_link_1" reason="Adjacent" />
    <disable_collisions link1="ar4_link_1" link2="ar4_link_2" reason="Adjacent" />
    <disable_collisions link1="ar4_link_2" link2="ar4_link_3" reason="Adjacent" />
    <disable_collisions link1="ar4_link_3" link2="ar4_link_4" reason="Adjacent" />
    <disable_collisions link1="ar4_link_4" link2="ar4_link_5" reason="Adjacent" />
    <disable_collisions link1="ar4_link_5" link2="ar4_link_6" reason="Adjacent" />
    <disable_collisions link1="ar4_link_5" link2="ar4_gripper_base_link" reason="Adjacent" />
    <disable_collisions link1="ar4_link_6" link2="ar4_gripper_base_link" reason="Adjacent" />
    <disable_collisions link1="ar4_gripper_base_link" link2="ar4_gripper_jaw1_link" reason="Adjacent" />
    <disable_collisions link1="ar4_gripper_base_link" link2="ar4_gripper_jaw2_link" reason="Adjacent" />

    <disable_collisions link1="link_6" link2="flange" reason="Adjacent" />
    <disable_collisions link1="flange" link2="tool0" reason="Adjacent" />
    <disable_collisions link1="link_6" link2="tool0" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="tool0" reason="Adjacent" />
    <disable_collisions link1="gripper_base_link" link2="flange" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="link_6" reason="Never" />
    <disable_collisions link1="gripper_ABB_Gripper_Finger_1" link2="gripper_base_link" reason="Adjacent" />
    <disable_collisions link1="gripper_ABB_Gripper_Finger_2" link2="gripper_base_link" reason="Adjacent" />
    
    <disable_collisions link1="abb_table" link2="ar4_table" reason="Adjacent" />
    <disable_collisions link1="abb_table" link2="camera" reason="Adjacent" />
    <disable_collisions link1="ar4_table" link2="camera" reason="Never" />
    <disable_collisions link1="abb_table" link2="base_link" reason="Adjacent" />
    <disable_collisions link1="ar4_table" link2="ar4_base_link" reason="Adjacent" />

    <disable_collisions link1="ar4_gripper_jaw1_link" link2="ar4_gripper_jaw2_link" reason="Default" />
    <disable_collisions link1="gripper_ABB_Gripper_Finger_1" link2="gripper_ABB_Gripper_Finger_2" reason="Default" />

</robot>